File: mpdcurrentsong.rb
#!/usr/bin/env ruby
require 'rubygems'                              # Rubygems
require 'librmpd'                               # MPD Library
require 'xosd'                                  # OSD Library

mpd = MPD.new 'localhost', 6600                 # Init MPD
mpd.connect                                     # Connect to MPD Server
song = mpd.current_song                         # Get the current song

osd = XOSD::Xosd.new(5)                         # Init XOSD
osd.font = "-adobe-courier-*-r-*-*-24"          # Choose XOSD Font
osd.color = "#287D28"                           # Choose XOSD color (dark green)

if song.title then                              # Check to see if the song has a title
  osd.display_message(3, song.title)            # If so , display the title
  sleep(2)                                      # Display it for 2 seconds
else                                            
  osd.display_message(3, song.file)             # if there is no title, display the filename
  sleep(2)                                      # Display it for 2 seconds
end

File: mpdcurrentsong.py
#!/usr/bin/env python
import mpd 					                        # Import MPD Library
import pyosd										            # Import OSD Library
import time											            # Import Internal Time Module

client = mpd.MPDClient()							      # Init MPD Client
client.connect("localhost", 6600)					  # Connect to local MPD Server

cs = client.currentsong()                   # Get the currentsong dict
if 'title' in cs:                           # Check to see if "title" title exists in the dict
 	songtitle = cs['title']                   # If it does set songtitle to the id3 title
elif 'file' in cs:                          # If it doesn't have a title use the filename
 	songtitle = cs['file']                    # Set songtitle to the filename

songartist = " by "
songartist += cs['artist']                  # Set songartist if the song has one

FONT = "-*-helvetica-*-r-normal--34-*-*"    # Set the font (a big bold one that's clearly visible)

COLOR = "#287D28"									        # Set the color (Dark Gren)
p = pyosd.osd(font=FONT, colour=COLOR)		  # init pyosd with the font and color
p.display(songtitle  + songartist)          # Finally use Pyosd to display the song title

time.sleep(2)                               # Show the message for 2 seconsd
